module/state_synchronization/indexer/collection_executed_metric.go

```diff
@@ -87,17 +95,41 @@ func (c *CollectionExecutedMetricImpl) BlockFinalized(block *flow.Block) {
 		}
 
 		for _, t := range l.Transactions {
+			c.blockTransactions[blockID] = append(c.blockTransactions[blockID], t)
```

Comment:
now that you are reading and writing to a vanilla map, you'll need to add some locking to this struct to prevent concurrent read/write